pkg/notifier/ocmnotifier.go

```diff
@@ -95,46 +119,48 @@ func (s *ocmNotifier) getPolicyIdForUpgradeConfig(clusterId string) (*string, er
 }
 
 // Validates that a state transition can be made from the supplied from/to states
-func validateStateTransition(from NotifyState, to NotifyState) bool {
+func validateStateTransition(from OcmState, to OcmState) bool {
 
 	switch from {
-	case StatePending:
+	case OcmStatePending:
 		// We shouldn't even be in this state to transition from
 		return false
-	case StateScheduled:
+	case OcmStateScheduled:
 		// Can only go to a started state
 		switch to {
-		case StateStarted:
+		case OcmStateStarted:
 			return true
 		default:
 			return false
 		}
-	case StateStarted:
+	case OcmStateStarted:
 		// Can go to a delayed, completed or failed state
 		switch to {
-		case StateDelayed:
+		case OcmStateDelayed:
 			return true
-		case StateCompleted:
+		case OcmStateCompleted:
 			return true
-		case StateFailed:
+		case OcmStateFailed:
 			return true
 		default:
 			return false
 		}
-	case StateDelayed:
+	case OcmStateDelayed:
 		// can go to completed or failed state
 		switch to {
-		case StateCompleted:
+		case OcmStateCompleted:
+			return true
+		case OcmStateFailed:
 			return true
-		case StateFailed:
+		case OcmStateDelayed:
```

Comment:
Have to add this here, since it will change from a delay to another delay.
The problem here is if we do delayerTrigger and the scaleTimeout have different values, it will send the delay notification every reconcile loop.

Reply:
I agree this is a problem, and I've been able to replicate it not just with scale timeout logs but with delays due to health check whilst testing the PR.

Maybe we can rework this function to be based around `MuoState` rather than `OcmState`?

It would be similar rules except:

- a MuoStateDelayed can transition to a MuoStateDelayed but cannot transition to a MuoStateDelayed
- a MuoStateSkiped can transition to a MuoStateDelayed but cannot transition to MuoStateSkipped.

That doesn't solve 100% of the issues (can still flap between delayed -> skipped and skipped -> delayed) but might cut down on at least some of the potential for repetition.

Reply:
In that case, maybe we can just disable the transition from skipped to delay?

But there is one problem for this, when sending the notification in `func (s *ocmNotifier) NotifyState`, we need to fetch the current OCM state. Then we need to convert this back to Muo state.

The problem is the state map has two values matched to `OcmStateDelayed`, I think if we write a new function to convert the OCM state to Muo state, it will return a random value from MuoStateSkipped and MuoStateDelayed.

WDYT?